src/ecs/resources/input.rs

```diff
@@ -433,82 +390,118 @@ impl InputHandler {
+        self.actions
+            .get(action.as_ref())
+            .map(|ref buttons| buttons.iter().any(|&b| self.button_down(b)))
     }
 
     /// Checks if the given action was released on this frame.
-    pub fn action_released(&self, action: i32) -> Option<bool> {
-        self.actions.get(&action).map(|&b| self.button_released(b))
+    pub fn action_released<T: AsRef<str>>(&self, action: T) -> Option<bool> {
+        self.actions
+            .get(action.as_ref())
+            .map(|ref buttons| buttons.iter().any(|&b| self.button_released(b)))
     }
 
-    /// Checks if the all the given actions are being pressed and at least one was pressed this frame.
+    /// Checks if the all given actions are being pressed and at least one was pressed this frame.
     ///
     /// If any action in this list is invalid this will return the id of it in Err.
-    pub fn actions_down(&self, actions: &[i32]) -> Result<bool, Vec<i32>> {
-        let mut all_buttons_are_pressed = true;
-        let mut any_button_is_pressed_this_frame = false;
+    pub fn actions_down<T: AsRef<str>>(&self, actions: &[T]) -> Result<bool, Vec<String>> {
+        let mut all_actions_are_pressed = true;
+        let mut any_action_is_pressed_this_frame = false;
         let mut bad_values = Vec::new();
         for action in actions {
-            if let Some(button) = self.actions.get(action) {
-                if !any_button_is_pressed_this_frame && self.button_down(*button) {
-                    any_button_is_pressed_this_frame = true;
+            if let Some(buttons) = self.actions.get(action.as_ref()) {
+                if !any_action_is_pressed_this_frame {
+                    if buttons.iter().any(|&b| self.button_down(b)) {
+                        any_action_is_pressed_this_frame = true;
+                    }
                 }
-                if all_buttons_are_pressed && !self.button_is_pressed(*button) {
-                    all_buttons_are_pressed = false;
+                if all_actions_are_pressed {
+                    if buttons.iter().all(|&b| !self.button_is_pressed(b)) {
+                        all_actions_are_pressed = false;
+                    }
                 }
             } else {
-                bad_values.push(*action);
+                bad_values.push(action.as_ref().to_string());
             }
         }
         if !bad_values.is_empty() {
             Err(bad_values)
         } else {
-            Ok(all_buttons_are_pressed && any_button_is_pressed_this_frame)
+            Ok(all_actions_are_pressed && any_action_is_pressed_this_frame)
         }
     }
 
     /// Assign an axis to an ID value
     ///
     /// This will insert a new axis if no entry for this id exists.
     /// If one does exist this will replace the axis at that id and return it.
-    pub fn insert_axis(&mut self, id: i32, axis: Axis) -> Option<Axis> {
+    pub fn insert_axis(&mut self, id: String, axis: Axis) -> Option<Axis> {
```

Comment:
Sorry if it has been mentioned before, but why not be consistent and accept `Into<String>` here? I mean, this is the place where it would actually make the code shorter in most cases (because you most likely have a `&str`).